leader-election-impl/src/test/java/com/palantir/leader/LeadershipElectionCheckTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.leader;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.IntStream;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.codahale.metrics.Clock;
+import com.codahale.metrics.Meter;
+import com.palantir.leader.health.LeaderElectionHealthCheck;
+import com.palantir.leader.health.LeaderElectionHealthStatus;
+import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
+
+public class LeadershipElectionCheckTest {
+    private final FakeTimeClock fakeTimeClock = new FakeTimeClock();
+    private final TaggedMetricRegistry registry1 = mock(TaggedMetricRegistry.class);
+    private final LeaderElectionServiceMetrics leaderElectionServiceMetrics =
+            LeaderElectionServiceMetrics.of(registry1);
+    private final LeaderElectionHealthCheck leaderElectionHealthCheck =
+            new LeaderElectionHealthCheck(leaderElectionServiceMetrics);
+
+    @Before
+    public void setup() {
+        when(registry1.meter(any())).thenReturn(new Meter(fakeTimeClock));
+    }
+
+    @Test
+    public void shouldBeHealthyForOneLeaderElectionPerMinute() {
+        markLeaderElectionsAtSpecifiedInterval(3, 60);
+
+        assertThat(leaderElectionHealthCheck.leaderElectionRateHealthStatus())
+                .isEqualTo(LeaderElectionHealthStatus.HEALTHY);
+    }
+
+    @Test
+    public void shouldBeUnhealthyForMoreThanOneLeaderElectionPerMinute() {
+        markLeaderElectionsAtSpecifiedInterval(7, 30);
+        assertThat(leaderElectionHealthCheck.leaderElectionRateHealthStatus())
+                .isEqualTo(LeaderElectionHealthStatus.UNHEALTHY);
+    }
+
+    public void markLeaderElectionsAtSpecifiedInterval(int leaderElectionCount, long timeIntervalInSeconds) {
+        // First tick
+        fakeTimeClock.advance(6, TimeUnit.SECONDS);
```

Comment:
Sorry - despite the comment I'm not sure what the significance of the 6 seconds here is?